installer/Util/SystemConfig/SystemConfiguration.php

```diff
@@ -0,0 +1,421 @@
+        return $qb->select('COUNT(employee.emp_number) as employeeCount')
+            ->from('hs_hr_employee', 'employee')
+            ->fetchOne();
+    }
+
+    /**
+     * @param string $organizationName
+     * @param string $email
+     * @param string $adminFirstName
+     * @param string $adminLastName
+     * @param string $host
+     * @param string $country
+     * @param string $ohrmVersion
+     * @param int $currentTimestamp
+     */
+    public function setInstanceIdentifier(
+        string $organizationName,
+        string $email,
+        string $adminFirstName,
+        string $adminLastName,
+        string $host,
+        string $country,
+        string $ohrmVersion,
+        int $currentTimestamp
+    ): void {
+        $instanceIdentifier = $this->createInstanceIdentifier(
+            $organizationName,
+            $email,
+            $adminFirstName,
+            $adminLastName,
+            $host,
+            $country,
+            $ohrmVersion,
+            $currentTimestamp,
+        );
+        $this->configHelper->setConfigValue(
+            self::INSTANCE_IDENTIFIER,
+            $instanceIdentifier
+        );
+    }
+
+    /**
+     * @param string $organizationName
+     * @param string $email
+     * @param string $adminFirstName
+     * @param string $adminLastName
+     * @param string $host
+     * @param string $country
+     * @param string $ohrmVersion
+     * @param int $currentTimestamp
+     */
+    public function setInstanceIdentifierChecksum(
+        string $organizationName,
+        string $email,
+        string $adminFirstName,
+        string $adminLastName,
+        string $host,
+        string $country,
+        string $ohrmVersion,
+        int $currentTimestamp
+    ): void {
+        $instanceIdentifierChecksum = $this->createInstanceIdentifierChecksum(
+            $organizationName,
+            $email,
+            $adminFirstName,
+            $adminLastName,
+            $host,
+            $country,
+            $ohrmVersion,
+            $currentTimestamp
+        );
+        $this->configHelper->setConfigValue(
+            self::INSTANCE_IDENTIFIER_CHECKSUM,
+            $instanceIdentifierChecksum
+        );
+    }
+
+    /**
+     * @param string $organizationName
+     * @param string $email
+     * @param string $adminFirstName
+     * @param string $adminLastName
+     * @param string|null $host
+     * @param string|null $country
+     * @param string $ohrmVersion
+     * @param int $currentTimestamp
+     * @return string
+     */
+    private function createInstanceIdentifier(
+        string $organizationName,
+        string $email,
+        string $adminFirstName,
+        string $adminLastName,
+        ?string $host,
+        ?string $country,
+        string $ohrmVersion,
+        int $currentTimestamp
+    ): string {
+        $host = !is_null($host) ?: '';
+        $country = !is_null($country) ?: '';
```

Comment:
is this correct? this basically means
```
        $host = !is_null($host) ? !is_null($host) : '';
        $country = !is_null($country) ? !is_null($host) : '';
```
right? Not sure if intention is to get boolean values for variables or not

Reply:
yes this is a short hand syntax we can use if we are using conditional value for true return.
$host = !is_null($host) ?: '' is identical to   $host = !is_null($host) ? !is_null($host) : '';
These are IDE refactoring sugessions

Reply:
https://www.phptutorial.net/php-tutorial/php-ternary-operator/#:~:text=HTML%2C%20XML%20(xml)-,The%20shorthand%20ternary%20operator,-Starting%20from%20PHP

Reply:
@SandeepaOHRM seems logic wrong.
I think this should be correct as;
```php
        $host = !is_null($host) ? $host : '';
        $country = !is_null($country) ? $country : '';
```
or
```php
        $host = $host ?? '';
        $country = $country ?? '';
```
or
```php
        return base64_encode(
            $organizationName .
            '_' . $email .
            '_' . $adminFirstName .
            '_' . $adminLastName .
            '_' . ($host ?? '') .
            '_' . ($country ?? '') .
            '_' . $ohrmVersion .
            '_' . $currentTimestamp
        );
```

Reply:
OMG. Fixed this. But not commited. Will commit